### ISSUE TITLE
Clarification about inpaint_loss

### ISSUE THREAD

**Issue by JLC7** [moodoki/semantic_image_inpainting#8]
Regarding the calculation of the inpaint_loss, it is unclear to me why you are summing the context_loss, which is a vector of length batch_size, with perceptual_loss which is a scalar (averaged for the batch) and not the individual losses.

`self.perceptual_loss = self.gl
self.inpaint_loss = self.context_loss + self.l*self.perceptual_loss`

Thank you

**Comment by moodoki** [comment]
Since we are processing a batch, this works out to be identical. When processing a batch, we are always optimizing batch loss. Please feel free to experiemnt with batch size of 1, however our pretrained graph does not allow for configurable batch sizes.
